### ISSUE TITLE
Knex columninfo typing return doesn't give correct type when no are columns provided

### ISSUE THREAD

**Issue by pulli23** [knex/knex#4228]
# Environment

Knex version: 0.21.1
Database + version: postgres
OS: ubuntu

Select applicable template from below.
If issue is about oracledb support, tag @ atiertant. For MSSql tag @ smorey2.
Rest of dialects don't need tags.
If issue is about TypeScript definitions, tag @ lorefnon.
@lorefnon 

# Bug

1. Explain what kind of behaviour you are getting and how you think it should do

When calling `Knex('table').columnInfo();` one expects the return value to be `Promise<{[columnName: string]: Knex.ColumnInfo}>`

2. Error message
```ColumnInfo is not assignable to {[columnName: string]: ColumnInfo```

3. Reduced test code, for example in https://npm.runkit.com/knex or if it needs real
   database connection to MySQL or PostgreSQL, then single file example which initializes
   needed data and demonstrates the problem.
Not applicable: this is a direct consequence of the way line 1616 in `index.d.ts` is written:
```
    columnInfo(column?: keyof TRecord): Promise<ColumnInfo>;
```

Can be fixed by changing the line to:

```
    columnInfo(column: keyof TRecord): Promise<ColumnInfo>;
    columnInfo(): Promise<{[column: string]: ColumnInfo}>;
```

**Comment by kibertoad** [comment]
Released in 1.0.1.

**Comment by timbrownsf** [comment]
Also the ColumnInfo type is wrong - maxLength should be nullable
